client/state/purchases/selectors.js

```diff
@@ -70,6 +71,22 @@ export const getByPurchaseId = ( state, purchaseId ) =>
 export const getSitePurchases = ( state, siteId ) =>
 	getPurchases( state ).filter( ( purchase ) => purchase.siteId === siteId );
 
+/**
+ * Returns a list of Purchases associated with a Site that may be expiring soon
+ * or have expired recently but is still renewable.
+ *
+ * @param {object} state      global state
+ * @param {number} siteId     the site id
+ * @returns {Array} the matching expiring purchases if there are some
+ */
+export const getRenewableSitePurchases = ( state, siteId ) =>
+	getSitePurchases( state, siteId ).filter(
+		( purchase ) =>
+			( isExpiring( purchase ) || isExpired( purchase ) ) &&
+			canExplicitRenew( purchase ) &&
+			moment( purchase.expiryDate ).diff( Date.now(), 'days' ) < 90
```

Comment:
@DavidRothstein these are the current conditions being checked to determine the purchases that qualify as "upcoming renewal", do you think I should remove the `isExpiring`,`isExpired` checks to include active subscriptions as well (the ones with auto-renew = on) ?

Reply:
Curious what others think, but my gut reaction is yes. That it's better to take the renewals together in a single transaction while the user is here, both for us and the user. We just want to avoid giving the user a false impression. Maybe if it's still active it says "renews in 3 days" instead of "expires in 3 days"?